src/components/profile/Profile.js

```diff
@@ -266,6 +286,18 @@ export function Profile({ match }) {
                                 )}
                             </>
                         }
+                        {!account.ledgerKey && !isMobile &&
+                            <MobileSharing
+                                onSetMobileSharingLink={ async (option) => {
```

Comment:
Man, Profile is one scary place :D

Feels like:
1. The code here that is responsible for building `mobileSharingLink` should exist in a `MobileSharingWrapper` component  that has `useDispatch()` and whatever hook(s) we need from `react-router` (e.g. `useLocation`, `useHistory`, `useMatchedRoute`, `useParams` (reference: https://reactrouter.com/web/api/Hooks) :)
2. We can compute `mobileSharingLink` in the wrapper, in a `useEffect()` hook with the router location and `accountId` as its dependencies, and stash the result of the computation in a `mobileSharingLink` `useState()` variable in the wrapper (like you are doing here now), so that we always re-compute the link state any time either the location or the `accountId` changes (account switch safe!)...
3. Then, I think we no longer need any lifecycle to `clear` the link because it is always valid; all we need to do is show or hide the modal that _displays_ the link :) I think you could completely remove the `onSetMobileSharingLink` prop from the `MobileSharing` component entirely at that point

Whaddya think? It also means that we will no longer be doing an async operation to compute the link _as we're showing the modal_ which might be a bit weird  and cause a flicker in some cases